frontend/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,105 @@
+# Code Guidelines for React
+
+Hat tip to Clean Code by Robert C. Martin (Uncle Bob)
+
+## Goals
+
+- Consistency
+- Predictability
+- Maintainability
+- Readability
+
+## Style
+
+- Take time to craft great variable and function names. It's ok to be verbose. Avoid abbreviations.
+- Replace magic numbers with named constants.
+- Remove dead code aggressively. We can always recover it from version control.
+- Use vertical whitespace to aid readability.
+  - Related lines should be vertically close.
+  - A block of variable declarations should be followed by a blank line.
+  - Use a blank line above return statements.
+  - Don't stack code too tight vertically, especially `if` blocks.
+- Use Prettier/ESLint to avoid bike-shedding.
+- Prefer copy/paste to premature abstraction.
+  - Abstractions tend to reveal themselves quickly on the second or third copy/paste.
+  - It's easier to abstract code than to remove an abstraction.
+- Break switch statements into their own functions with returns rather than `break;`.
+
+## File structure
+
+- Higher-order functions go at the top of the file when possible.
+- Named exports are preferred to default exports. It's easier to control naming.
+- Inline exports are preferred to the define-then-export pattern.
```

Comment:
I sort of like having all the exports at the bottom in one statement, but could see this making it easy to know what is private and public inline. I think explicitly making exports at the bottom could reduce the amount of methods that are exported based on copying/pasting another method.

```
const sum = (number1,  number2) => printNumber(number1 + number2)

const subtract = (minuend, subtrahend) => printNumber(minuend - subtrahend)

const printNumber = (number) => console.log(number)

export { sum, subtract }
```

Reply:
That's a point I hadn't considered.

I do a lot of `export function SomeComponent(){}`, so most files start out with a single top-level export and then might get more exports if necessary. I wonder if this is the `function` keyword pattern vs the `const` function assignment pattern.